tests/integration/data_sources_and_expectations/expectations/test_expect_column_distinct_values_to_be_in_set.py

```diff
@@ -0,0 +1,108 @@
+from datetime import datetime
+
+import pandas as pd
+import pytest
+
+import great_expectations.expectations as gxe
+from great_expectations.core.result_format import ResultFormat
+from great_expectations.datasource.fluent.interfaces import Batch
+from tests.integration.conftest import parameterize_batch_for_data_sources
+from tests.integration.data_sources_and_expectations.test_canonical_expectations import (
+    ALL_DATA_SOURCES,
+    JUST_PANDAS_DATA_SOURCES,
+)
+
+COL_NAME = "my-col"
+
+ONES_AND_TWOS = pd.DataFrame({COL_NAME: [1, 2, 2, 2]})
+
+
+@parameterize_batch_for_data_sources(data_source_configs=ALL_DATA_SOURCES, data=ONES_AND_TWOS)
+def test_success_complete_results(batch_for_datasource: Batch) -> None:
+    expectation = gxe.ExpectColumnDistinctValuesToBeInSet(column=COL_NAME, value_set=[1, 2])
+    result = batch_for_datasource.validate(expectation, result_format=ResultFormat.COMPLETE)
+    assert result.success
+    assert result.to_json_dict()["result"] == {
```

Comment:
This is ugly, but we need to call `to_json_dict()` to deal with some dataframe comparison issues.